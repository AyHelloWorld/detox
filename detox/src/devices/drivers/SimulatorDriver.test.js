jest.mock('../registries/SimulatorDeviceRegistry');
jest.mock('../../utils/logger');
jest.mock('../../utils/environment', () => ({
  getFrameworkPath: () => '',
  getDeviceLockFilePath: () => '',
}));

describe('SimulatorDriver', () => {
  let log;
  let SimulatorDriver;

  beforeEach(() => {
    log = require('../../utils/logger');
    SimulatorDriver = require('./SimulatorDriver');
  });

  it('should throw error if detox framework path has not been found', async () => {
    const driver = new SimulatorDriver({ client: null });

    await expect(driver.prepare()).rejects.toThrowErrorMatchingSnapshot();
  });

  it('should log error to log if cannot acquire free device from the registry', async () => {
    const driver = new SimulatorDriver({ client: null });
    driver.deviceRegistry.acquireDevice = jest.fn();

    await driver.acquireFreeDevice('test string');
    expect(log.error.mock.calls).toMatchSnapshot();
  });
});
