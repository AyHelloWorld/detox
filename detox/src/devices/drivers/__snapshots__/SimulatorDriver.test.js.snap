// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimulatorDriver should log error to log if cannot acquire free device from the registry 1`] = `
Array [
  Array [
    Object {
      "event": "ACQUIRE_DEVICE_ERROR",
    },
    "Unable to acquire free device: test string",
  ],
]
`;

exports[`SimulatorDriver should throw error if detox framework path has not been found 1`] = `
"DetoxRuntimeError:  could not be found, this means either you changed a version of Xcode or Detox postinstall script was unsuccessful.

HINT: To attempt a fix try running 'detox clean-framework-cache && detox build-framework-cache'"
`;
